node-hub/dora-mujoco-husky/dora_mujoco_husky/main.py

```diff
@@ -0,0 +1,84 @@
+from dora import Node
+import mujoco
+import mujoco.viewer
+import pyarrow as pa
+import os
+from .mesh_downloader import ensure_meshes
+
+def clamp(value, min_val, max_val):
+    """Clamp value between min_val and max_val"""
+    return max(min(value, max_val), min_val)
+
+def main():
+    try:
+        # Initialize node
+        node = Node("mujoco_husky")
+        print("Initializing Mujoco Husky simulation...")
+        
+        ensure_meshes()
+        
+        # Load Husky model
+        model_path = os.path.join(os.path.dirname(__file__), "husky/husky.xml")
+        model = mujoco.MjModel.from_xml_path(model_path)
+        data = mujoco.MjData(model)
+        
+        # Define velocity limits
+        MAX_LINEAR_VELOCITY = 2.0   
+        MAX_ANGULAR_VELOCITY = 3.0  
+        
+        # Initialize viewer
+        with mujoco.viewer.launch_passive(model, data) as viewer:
+            print("Simulation initialized successfully")
+
+            for event in node:
+                if event["type"] == "INPUT" and event["id"] == "cmd_vel":
+                    try:
```

Comment:
It seems to me that this try is very big although a lot of the code should never fail